diff --git a/dist/compiler/runtime/get-locale.js b/dist/compiler/runtime/get-locale.js
index afe0c8a8caf5347cc7987f20e5cead62b58cc473..9251120368ba5f8089b40bbe3b6c8141ffe0c53f 100644
--- a/dist/compiler/runtime/get-locale.js
+++ b/dist/compiler/runtime/get-locale.js
@@ -112,7 +112,8 @@ function resolveLocaleWithStrategies(strategyToUse, urlForUrlStrategy) {
         }
         else if (TREE_SHAKE_LOCAL_STORAGE_STRATEGY_USED &&
             strat === "localStorage" &&
-            !isServer) {
+            !isServer &&
+            typeof localStorage !== "undefined") {
             locale = localStorage.getItem(localStorageKey) ?? undefined;
         }
         else if (isCustomStrategy(strat) && customClientStrategies.has(strat)) {
